tests/unit/classifier/svm/NewtonSVM_unittest.cc

```diff
@@ -0,0 +1,133 @@
+*/
+#include <functional>
+#include <rxcpp/rx-lite.hpp>
+#include <shogun/lib/Signal.h>
+
+#include "environments/LinearTestEnvironment.h"
+#include <gtest/gtest.h>
+#include <shogun/base/some.h>
+#include <shogun/classifier/svm/NewtonSVM.h>
+#include <shogun/evaluation/ContingencyTableEvaluation.h>
+#include <shogun/features/DenseFeatures.h>
+#include <shogun/labels/BinaryLabels.h>
+
+using namespace shogun;
+
+extern LinearTestEnvironment* linear_test_env;
+
+TEST(NewtonSVM, train)
+{
+	auto env = linear_test_env->getBinaryLabelData();
+	auto features = wrap(env->get_features_train());
+	auto labels = wrap(env->get_labels_train());
+	auto test_features = wrap(env->get_features_test());
+	auto test_labels = wrap(env->get_labels_test());
+
+	auto svm = some<CNewtonSVM>();
+	svm->set_labels(labels);
+	EXPECT_TRUE(svm->train(features));
+	auto results = wrap(svm->apply(test_features));
+	auto acc = some<CAccuracyMeasure>();
+	EXPECT_EQ(acc->evaluate(results, test_labels), 1.0);
+}
+
+TEST(NewtonSVM, continue_training_consistency)
+{
+	auto env = linear_test_env->getBinaryLabelData();
+	auto features = wrap(env->get_features_train());
+	auto labels = wrap(env->get_labels_train());
+	auto test_features = wrap(env->get_features_test());
+	auto test_labels = wrap(env->get_labels_test());
+
+	auto svm = some<CNewtonSVM>();
+	svm->set_labels(labels);
+	svm->train(features);
+
+	// reference for completly trained model
+	auto results = svm->apply_binary(test_features);
+
+	index_t iter = 0;
+
+	// prematurely stopped at 5 iterations
+	auto svm_stop = some<CNewtonSVM>();
+
+	std::function<bool()> callback = [&iter]() {
+		if (iter >= 5)
+		{
+			get_global_signal()->get_subscriber()->on_next(SG_BLOCK_COMP);
+			return true;
+		}
+		iter++;
+		return false;
+	};
+	svm_stop->set_callback(callback);
+
+	svm_stop->set_compute_bias(false);
+	svm_stop->set_labels(labels);
+	svm_stop->train(features);
+
+	// callback executes, model should not be converged
+	ASSERT(!svm_stop->is_complete());
+
+	// reference model for intermediate state
+	auto svm_one = some<CNewtonSVM>();
+	svm_one->set_labels(labels);
+
+	// trained only till 5 iteraetions
+	svm_one->put<int32_t>("max_iterations", 5);
+	svm_one->set_compute_bias(false);
+	svm_one->train(features);
+
+	auto results_one = svm_one->apply_binary(test_features);
+	auto results_stop = svm_stop->apply_binary(test_features);
+
+	// prematurely stopped model and intermediate reference model should be
+	// consistent
+	EXPECT_TRUE(results_one->get_labels().equals(results_stop->get_labels()));
+
+	svm_stop->set_callback(nullptr);
+
+	// continue training till converged
+	svm_stop->continue_train();
+	auto results_complete = svm_stop->apply_binary(test_features);
+
+	// compare model with completely trained reference
+	EXPECT_TRUE(results_complete->get_labels().equals(results->get_labels()));
+
+	SG_UNREF(results_one);
+	SG_UNREF(results_stop);
+	SG_UNREF(results);
+	SG_UNREF(results_complete);
```

Comment:
no memory leaks/erriors in here?

Reply:
?

Reply:
there should not be since i only introduced bias change but i will check again